README.md

```diff
@@ -98,44 +98,95 @@ Usage: gpgbridge.rb [options]
 
 ## Example bash helper functions
 
-I have the following in my `~/.bash_profile` in WSL. Note that there is a
-line in start_gpgbridge that must be uncommented for WSL2.
+Copy the script below to somewhere accessible from WSL, make it 
+executable (`chmod +x path/to/gpgbridge_helper.sh`).
+
+Then add `source path/to/gpgbridge_helper.sh` to your `~/.bash_profile`, `~/.bashrc` or `~/.zshrc`.
+
+Make sure to edit the variables in the start of the script according to your
+setup.
 
 ```bash
+#!/usr/bin/env bash
 #--------------------------------------------------------------------------
 # GPG bridging from WSL gpg to gpg4win gpg-agent.exe
 # (needed to use a Yubikey, since WSL cannot access USB devices)
 
-if [[ -f /mnt/c/Program1/gpgbridge.rb ]]
+# Set this to true if running WSL1
+WSL2=false
+
+SCRIPT_DIR_WSL='/mnt/c/Program1/'
+# shellcheck disable=SC1003
+SCRIPT_DIR_WIN='C:\\Program1\\'
+
+PIDFILE_WSL="$HOME/.gpgbridge.pid"
+LOGFILE_WSL="$HOME/.gpgbridge.log"
+
+PIDFILE_WIN="${SCRIPT_DIR_WIN}gpgbridge.pid"
+LOGFILE_WIN="${SCRIPT_DIR_WIN}gpgbridge.log"
+
+SCRIPT_FILE_NAME='gpgbridge.rb'
+
+SCRIPT_PATH_WSL="${SCRIPT_DIR_WSL}${SCRIPT_FILE_NAME}"
+
+# Set to true to redirect output to /dev/null
+QUIET=false
+
+function start_gpgbridge
+{
+    if ! command -v ruby.exe >/dev/null
+    then
+        echo 'No ruby.exe found in path'
+        return
+    fi
+
+    local cmd=( \
+        'ruby' \
+        "$SCRIPT_PATH_WSL" \
+        '--verbose' \
+        '--daemon' \
+        "--pidfile $PIDFILE_WSL" \
+        "--logfile $LOGFILE_WSL" \
+        "--windows-pidfile $PIDFILE_WIN" \
+        "--windows-logfile $LOGFILE_WIN" \
+    )
+
+    if [ "$1" = 'ssh' ]
+    then
+        cmd+=('--enable-ssh-support')
+        SSH_AUTH_SOCK=$(gpgconf --list-dirs agent-ssh-socket)
+        export SSH_AUTH_SOCK
+    fi
+
+    if [ "$WSL2" = true ]
+    then
+        cmd+=("--remote-address $(ip route | awk '/^default via / {print $3}')")
+    fi
+
+    if [ "$QUIET" = true ]
+    then
+        cmd+=('>/dev/null 2>&1')
+    fi
+
+    printf -v _cmd '%s ' "${cmd[@]}"
+
+    eval "$_cmd"
```

Comment:
I'm thinking the reason you are using `eval` here is so that the redirection to /dev/null will work. Am I right that you don't want to see the message about an existing gpg bridge already running? Maybe it would be better to add a `--quiet` option to the script to make it silent.

Reply:
I'm using eval because I ran into issues where ruby would either complain about not finding a file named "path/to/bridge arg1 arg2 arg3" or not recognized an argument because it was single-quoted..

Reply:
I solved the quiet part with a lazy env variable, I didn't find a good way to parse non positional arguments.

Reply:
OK!